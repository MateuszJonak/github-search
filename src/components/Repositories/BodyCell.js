/* @flow */

import React from 'react';
import { get } from 'lodash/fp';
import TableCell from '@material-ui/core/TableCell';
import { type Column } from '../../lib/columns';

const identity = ({ value }) => value;

class BodyCell extends React.Component<{ column: Column, row: Object }> {
  render() {
    const { column, row } = this.props;
    const valueFormatter = column.valueFormatter || identity;
    const valueRenderer = column.valueRenderer || identity;

    return (
      <TableCell
        component={column.component}
        scope={column.component}
        numeric={column.numeric}>
        {valueRenderer({
          value: valueFormatter({
            value: get(column.path, row),
            row,
            ...this.props,
          }),
          row,
          ...this.props,
        })}
      </TableCell>
    );
  }
}

export default BodyCell;
