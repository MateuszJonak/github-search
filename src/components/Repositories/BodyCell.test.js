import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import BodyCell from './BodyCell';

describe('BodyCell', () => {
  let props;
  let tree;
  const shallowEnzyme = properties => shallow(<BodyCell {...properties} />);

  beforeEach(() => {
    props = {
      column: {
        id: 'id',
        path: 'id',
        label: 'ID',
      },
      row: {
        id: 1,
        name: 'name1',
        profile: {
          firstName: 'My name',
        },
      },
    };
    tree = shallowEnzyme(props);
  });

  it('should render component correctly with default props', () => {
    expect(toJson(tree.dive())).toMatchSnapshot();
  });

  it('should render component correctly with nested key', () => {
    tree.setProps({
      column: {
        id: 'firstName',
        path: 'profile.firstName',
        label: 'First name',
      },
    });
    expect(toJson(tree.dive())).toMatchSnapshot();
  });

  it('should render component correctly with some valueFormatter and valueRenderer props', () => {
    tree.setProps({
      column: {
        id: 'firstName',
        path: 'profile.firstName',
        label: 'First name',
        valueFormatter: ({ value }) => `${value}2`,
        valueRenderer: ({ value }) => <span>{value}</span>,
      },
    });
    expect(toJson(tree.dive())).toMatchSnapshot();
  });
});
