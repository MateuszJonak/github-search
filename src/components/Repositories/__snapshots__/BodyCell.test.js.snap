// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BodyCell should render component correctly with default props 1`] = `
<TableCell
  classes={
    Object {
      "body": "MuiTableCell-body-3",
      "footer": "MuiTableCell-footer-4",
      "head": "MuiTableCell-head-2",
      "numeric": "MuiTableCell-numeric-5",
      "paddingCheckbox": "MuiTableCell-paddingCheckbox-7",
      "paddingDense": "MuiTableCell-paddingDense-6",
      "paddingNone": "MuiTableCell-paddingNone-8",
      "root": "MuiTableCell-root-1",
    }
  }
  numeric={false}
  padding="default"
>
  1
</TableCell>
`;

exports[`BodyCell should render component correctly with nested key 1`] = `
<TableCell
  classes={
    Object {
      "body": "MuiTableCell-body-3",
      "footer": "MuiTableCell-footer-4",
      "head": "MuiTableCell-head-2",
      "numeric": "MuiTableCell-numeric-5",
      "paddingCheckbox": "MuiTableCell-paddingCheckbox-7",
      "paddingDense": "MuiTableCell-paddingDense-6",
      "paddingNone": "MuiTableCell-paddingNone-8",
      "root": "MuiTableCell-root-1",
    }
  }
  numeric={false}
  padding="default"
>
  My name
</TableCell>
`;

exports[`BodyCell should render component correctly with some valueFormatter and valueRenderer props 1`] = `
<TableCell
  classes={
    Object {
      "body": "MuiTableCell-body-3",
      "footer": "MuiTableCell-footer-4",
      "head": "MuiTableCell-head-2",
      "numeric": "MuiTableCell-numeric-5",
      "paddingCheckbox": "MuiTableCell-paddingCheckbox-7",
      "paddingDense": "MuiTableCell-paddingDense-6",
      "paddingNone": "MuiTableCell-paddingNone-8",
      "root": "MuiTableCell-root-1",
    }
  }
  numeric={false}
  padding="default"
>
  <span>
    My name2
  </span>
</TableCell>
`;
